src/integer/z/sample/uniform.rs

```diff
@@ -39,18 +39,82 @@ impl Z {
     /// - Returns a [`MathError`] of type [`InvalidInterval`](MathError::InvalidInterval)
     /// if the given `upper_bound` isn't at least bigger than `lower_bound + 1`,
     /// i.e. the interval size is at most `1`.
-    pub fn sample_uniform<T1, T2>(lower_bound: &T1, upper_bound: &T2) -> Result<Self, MathError>
-    where
-        T1: Into<Z> + Clone,
-        T2: Into<Z> + Clone,
-    {
-        let lower_bound: Z = lower_bound.clone().into();
-        let upper_bound: Z = upper_bound.clone().into();
+    pub fn sample_uniform(
+        lower_bound: impl Into<Z>,
+        upper_bound: impl Into<Z>,
+    ) -> Result<Self, MathError> {
+        let lower_bound: Z = lower_bound.into();
+        let upper_bound: Z = upper_bound.into();
 
         let interval_size = &upper_bound - &lower_bound;
         let sample = sample_uniform_rejection(&interval_size)?;
         Ok(&lower_bound + sample)
     }
+
+    /// Chooses a prime [`Z`] instance uniformly at random
```

Comment:
Hmm, what do you think, should we return a dedicated error, if the range does not have any prime?
Can we even check something like that?